tests/generic/test_datacontainer.py

```diff
@@ -459,6 +460,54 @@ def test_subclass_preservation(self):
         )
         self.pl.pop('subclass')
 
+    def test_stub(self):
+        """Lazily loaded containers should contain only stubs and only force them when directly accessed."""
+
+        self.pl.to_hdf(self.hdf, "lazy")
+        ll = self.hdf["lazy"].to_object(lazy=True)
+        self.assertTrue(all(isinstance(v, HDFStub) for v in ll._store),
+                        "Not all values loaded as stubs!")
+
+        ll_repr = repr(ll)
+        self.assertTrue(all(isinstance(v, HDFStub) for v in ll._store),
+                        "Some stubs have been loaded after getting string repr of container!")
+
+        ll0 = ll[0]
+        self.assertTrue(all(isinstance(v, HDFStub) for v in ll0._store),
+                        "Recursive datacontainers not lazily loaded!")
+
+        self.assertEqual(ll[0].foo, self.pl[0].foo,
+                         "Lazily loaded list not equal to orignal list!")
+
+        self.assertTrue(not isinstance(ll._store[0], HDFStub),
+                        "Loaded value not stored back into container!")
+
+    def test_stub(self):
```

Comment:
I wanted to peek at codacy before approving and it actually caught something meaningful 😝 This is a redefinition of the test in 463!